variables.tf

```diff
@@ -9,23 +9,33 @@ variable "name" {
   description = "The name of the Azure Bastion Host."
 }
 
-variable "resource_group_name" {
+variable "resource_group_id" {
```

Comment:
I understand why you made this change, but for consistency we use name: https://azure.github.io/Azure-Verified-Modules/specs/tf/res/#id-rmfr3---category-composition---resource-groups. You can construct the ID with string interpolation or by using the provider defined function is azapi and the azapi_client_config data source for the sub id.